tests/integration/targets/mso_schema_site_service_graph/tasks/main.yml

```diff
@@ -781,3 +792,4 @@
       - rm_sg_again is not changed
       - rm_sg_again.current == {}
       - rm_sg_again.previous == {}
+    when: version.current.version is version('4.0', '<') # no change in NDO4.0 because site will already be present when template is defined
```

Comment:
You are having an issue on the remove again but not on the remove itself?

Reply:
Yes so the remove again will be the same output as the remove for version 4.0 and higher. Because when you remove the site_service_graph the template is still present and resets the site to the previous value. Thus it will always change and will not have a previous of {}